components/dada-ir/src/code/syntax.rs

```diff
@@ -98,8 +98,8 @@ pub enum ExprData {
     /// true, false
     BooleanLiteral(bool),
 
-    /// `22`, `22_222`, etc
-    IntegerLiteral(Word),
+    /// (`22i`, suffix_start: 2), (`22_222u`, suffix_start: 6), etc
+    IntegerLiteral(Word, usize),
```

Comment:
...then this could be `IntegerLiteral(Word, Word)`, too, which seems mildly nicer.